src/field.rs

```diff
@@ -181,6 +181,73 @@ pub fn summator_incompr(
     }
 }
 
+/// The Fourier method for scalar fields.
+///
+/// Computes the periodic, isotropic spatial random field $u(x)$ by the Fourier
+/// method according to
+/// $$
+/// u(x) = \sum_{i=1}^N \sqrt{2S(k_i\Delta k}\left(
+/// z_{1,i} \cdot \cos(\langle k_i, x \rangle) +
+/// z_{2,i} \cdot \sin(\langle k_i, x \rangle)\right)
+/// $$
+/// with
+/// * $S$ being the spectrum of the covariance model
+/// * $N$ being the number of Fourier modes
+/// * $z_1, z_2$ being independent samples from a standard normal distribution
+/// * $k$ being the equidistant Fourier grid
+/// and are given by the argument `modes`.
+/// * $\Delta k$ being the cell size of the Fourier grid
+///
+/// # Arguments
+///
+/// * `spectrum_factor` - the pre-calculated factor $\sqrt{2S(k_i\Delta k)}
+/// <br>&nbsp;&nbsp;&nbsp;&nbsp; dim = Fourier modes $N$
+/// * `modes` - equidistant Fourier grid, $k$ in Eq. above
+/// <br>&nbsp;&nbsp;&nbsp;&nbsp;dim = (spatial dim. of field $d$, Fourier modes $N$)
+/// * `z1` - independent samples from a standard normal distribution
+/// <br>&nbsp;&nbsp;&nbsp;&nbsp;dim = Fourier modes $N$
+/// * `z2` - independent samples from a standard normal distribution
+/// <br>&nbsp;&nbsp;&nbsp;&nbsp;dim = Fourier modes $N$
+/// * `pos` - the position $x$ where the spatial random field is calculated
+/// <br>&nbsp;&nbsp;&nbsp;&nbsp;dim = (spatial dim. of field $d$, no. of spatial points where field is calculated $j$)
+/// * `num_threads` - the number of parallel threads used, if None, use rayon's default
+///
+/// # Returns
+///
+/// * `summed_modes` - the isotropic spatial field
+/// <br>&nbsp;&nbsp;&nbsp;&nbsp;dim = no. of spatial points where field is calculated $j$
+pub fn summator_fourier(
+    spectrum_factor: ArrayView1<'_, f64>,
+    modes: ArrayView2<'_, f64>,
+    z1: ArrayView1<'_, f64>,
+    z2: ArrayView1<'_, f64>,
+    pos: ArrayView2<'_, f64>,
+    num_threads: Option<usize>,
+) -> Array1<f64> {
+    assert_eq!(modes.dim().0, pos.dim().0);
+    assert_eq!(modes.dim().1, z1.dim());
+    assert_eq!(modes.dim().1, z2.dim());
+
+    rayon::ThreadPoolBuilder::new()
+        .num_threads(num_threads.unwrap_or(rayon::current_num_threads()))
+        .build()
```

Comment:
This will add a significant cost for initializing a new thread pool for each and every call, i.e. via the thread start-up cost.

Maybe this could be changed to use the global/default thread pool if `num_threads` is `None`? Especially as `rayon::current_num_threads` is not the default (that would be `RAYON_NUM_THREADS` or `std::thread::available_parallelism` otherwise), but the number of threads in the global thread pool in contrast to the function documentation. (So basically it will use the number of threads chosen when the global pool was initialized which of course will be the default if it was initialized implicitly/automatically.)

Maybe something like

```rust
fn with_num_threads<F, R>(num_threads: Option<usize>, f: F) where F: FnOnce() -> R {
  if let Some(num_threads) = num_threads {
    ThreadPoolBuilder::new(num_threads).build().unwrap().install(f)
  } else {
    f()
  }
}
```